cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/cask360/Cask360Entity.java

```diff
@@ -0,0 +1,361 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.dataset.lib.cask360;
+
+import co.cask.cdap.api.dataset.lib.cask360.Cask360Group.Cask360GroupType;
+import co.cask.cdap.api.dataset.lib.cask360.Cask360GroupData.Cask360GroupDataMap;
+import co.cask.cdap.api.dataset.lib.cask360.Cask360GroupData.Cask360GroupDataTime;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import com.google.gson.JsonSyntaxException;
+
+import java.lang.reflect.Type;
+import java.text.ParseException;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Represents a single entity in a {@link Cask360Table}.
+ * <p>
+ * An entity is represented by a unique String id and any number of groups of
+ * data about the entity. Each group of data is identified by a String name and
+ * a {@link Cask360Group} object.
+ */
+public class Cask360Entity implements Comparable<Cask360Entity> {
```

Comment:
I prefer not to have company name as class name (we are following that since the last time we renamed from Continuuity). Maybe call it `User360` or `Profile360`?